server/src/main/java/io/crate/execution/engine/aggregation/impl/average/AverageAggregation.java

```diff
@@ -260,13 +164,13 @@ public AverageState newState(RamAccounting ramAccounting,
                                  Version indexVersionCreated,
                                  Version minNodeInCluster,
                                  MemoryManager memoryManager) {
-        ramAccounting.addBytes(AverageStateType.INSTANCE.fixedSize());
-        return new AverageState();
+        ramAccounting.addBytes(stateDateTypeSupplier.get().fixedSize());
+        return stateSupplier.get();
     }
 
     @Override
     public DataType<?> partialType() {
-        return AverageStateType.INSTANCE;
+        return (DataType<?>) stateDateTypeSupplier.get();
```

Comment:
See previous comment about making the type of the `DateTypeSupplier` more concrete. There shouldn't be a reason to have a cast here.

Reply:
Done. I created parent class for both types as I still needed cast to FixedWidthType when I did `ramAccounting.addBytes(((FixedWidthType) stateDateType).fixedSize());`